src/react-hooks-nesting-walker/error-messages.ts

```diff
@@ -17,4 +17,5 @@ export const ERROR_MESSAGES = {
     'A hook cannot be used inside of another function',
   invalidFunctionExpression: 'A hook cannot be used inside of another function',
   hookAfterEarlyReturn: 'A hook should not appear after a return statement',
+  anonymousFunctionIllegalCallback: `Hook is in an anonymous function that is passed to an illegal callback. Legal callbacks identifiers that can receive anonymous functions as arguments are memo and forwardRef`,
```

Comment:
Looking at this error message, I am afraid it may be confusing for regular developers who were not working with ASTs/language parsing before.

```suggestion
  anonymousFunctionIllegalCallback: `Hook is used in an anonymous function that is provided as an argument in an unexpected function. Functions that can receive anonymous functions with hook calls are "React.memo" and "React.forwardRef". If this is intended, add a name to the anonymous wrapping function to make it appear as a component.`,
```